src/promptflow/promptflow/_sdk/entities/_run.py

```diff
@@ -132,9 +139,14 @@ def __init__(
         self._creation_context = kwargs.get("creation_context", None)
         # init here to make sure those fields initialized in all branches.
         self.flow = flow
+        self._use_remote_flow = is_remote_uri(flow)
         self._experiment_name = None
         self._lineage_id = None
-        if self._run_source == RunInfoSources.LOCAL:
+        if self._use_remote_flow:
+            self._flow_name = get_flow_name_from_remote_flow_pattern(flow)
+            self._experiment_name = self._flow_name
```

Comment:
what's the _flow_name here when _use_remote_flow?

Reply:
It's the flow id aka flow name.